### DISCUSSION TITLE
Stitching multiple queries into 1, much like Relay

### DISCUSSION THREAD

**Discussion by paulm17** [urql-graphql/urql#2192] [Q&A] [answered]
So I'm new to urql and graphql in general.

When I hear about relay, apollo.  They talk about being able to concatinate multiple queries into 1 request.  I'm not sure if this is the right terminology to get my point across 😅.

Here is a slide from the latest relay presentation:  

![alt text](https://i.imgur.com/adZa0yY.png) 

Is it possible to do the same?  Currently I have about 6? hooks in play in my application and on init.  I see 6 requests going out.

Is it possible to reduce this to 1 request, keeping them as they are?  Like as relay mentions, they take the queries from all the components and writes out the best query?

Source: https://youtu.be/lhVGdErZuN4?t=180

Thanks

**Comment by JoviDeCroock** [comment, marked as answer]
Currently we aren't doing hoisting of queries, in Relay this happens through the compiler. However we can ensure that multiple queries can share data across queries through the [`normalized-cache`](https://formidable.com/open-source/urql/docs/graphcache/)
